webhooks/services/acapy_events_processor.py

```diff
@@ -338,6 +341,39 @@ def _process_event(self, event_json: str) -> None:
                 event_json=webhook_event_json, transaction_id=transaction_id
             )
 
+        if LAGO_API_KEY:
+            if is_applicable_for_billing(
+                wallet_id=wallet_id,
+                group_id=group_id,
+                topic=cloudapi_topic,
+                payload=payload,
+                logger=bound_logger,
+            ):
+                bound_logger.debug(
+                    "Forwarding billing event for Billing service",
+                )
+                if cloudapi_topic == "endorsements":
+                    # Add the operation type to the payload for endorsements
+                    # Simplifies the billing service's logic for determining the operation type
+
+                    operation_type = get_operation_type(
+                        payload=payload, logger=bound_logger
+                    )
```

Comment:
Note: `get_operation_type` is effectively called twice -- once in `is_applicable_for_billing`, and once again here ...

One possible optimisation is to make is_applicable_for_billing return a tuple ... i.e. the True/False result, but also the operation_type ... so we say:

```
if LAGO_API_KEY:
    is_applicable, operation_type = is_applicable_for_billing(
                wallet_id=wallet_id,
                group_id=group_id,
                topic=cloudapi_topic,
                payload=payload,
                logger=bound_logger,
            )
  if is_applicable:
    
    if cloudapi_topic == "endorsements":
      # we already have operation_type returned from the `is_applicable_for_billing` check, so just add to [payload][type]
```

Optimisation is the root of all evil, so just an idea! Because get_operation_type does a bunch of key extraction + json transforms